segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransaction.java

```diff
@@ -312,29 +312,30 @@ void preProcessOperation(Operation operation) throws ContainerException, StreamS
             if (segmentMetadata.isDeleted()) {
                 throw new StreamSegmentNotExistsException(segmentMetadata.getName());
             }
+            if (operation instanceof StreamSegmentAppendOperation) {
```

Comment:
I don't know why we need to do this. In the future, if one of these operations does not implement `SegmentOperation` anymore, this will easily lead to bugs. I would vote to undoing this change.

Reply:
If one of those classes does not implement SegmentOperation it will result in an NPE, which is what the warning was triggering on.